src/components/SearchableSelect/SearchableSelect.jsx

```diff
@@ -0,0 +1,427 @@
+import React from 'react';
+import _ from 'lodash';
+import { createClass, findTypes } from '../../util/component-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import { partitionText } from '../../util/text-manipulation';
+import * as reducers from './SearchableSelect.reducers';
+import CaretIcon from '../Icon/CaretIcon/CaretIcon';
+import DropMenu from '../DropMenu/DropMenu';
+import LoadingIcon from '../Icon/LoadingIcon/LoadingIcon';
+import SearchField from '../SearchField/SearchField';
+
+const cx = lucidClassNames.bind('&-SearchableSelect');
+
+const {
+	any,
+	bool,
+	func,
+	node,
+	number,
+	object,
+	shape,
+	string,
+	oneOfType,
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["controls", "selectors"], "madeFrom": ["DropMenu"]}
+ *
+ * A selector control (like native `<select>`) which is used to select a single option from a dropdown list.
+ * Supports option groups with and without labels.
+ */
+const SearchableSelect = createClass({
```

Comment:
I think part of the goal with this component is to address some of the concerns noted here: https://github.com/appnexus/lucid/issues/279 regarding async behavior.

I totally agree, though, that doing async loading of options with this component is risky territory.

When you say 'show the entire list at once', do you mean that you think that there shouldn't even be scrolling within the Flyout? Or just that the entire list of Options should be provided at mount and then never change?